src/jit-diff/jit-diff.cs

```diff
@@ -159,25 +184,240 @@ private void SetRID()
+                        return "Linux";
+                    case "Darwin":
+                        return "OSX";
+                    default:
+                        Console.Error.WriteLine("No platform mapping!");
+                        return null;
+                }
+            }
+
+            private void SetDefaults()
+            {
+                // If --diff_root wasn't specified, see if we can figure it out from the current directory
+                // using git. NOTE: we shouldn't do this unless it will be used, that is, if there are some
+                // arguments that are currently unspecified and need this to compute their default.
+                if (_diffRoot == null)
+                {
+                    _diffRoot = GetRepoRoot();
+                }
+
+                if ((_outputPath == null) && (_diffRoot != null))
+                {
+                    _outputPath = Path.Combine(_diffRoot, s_defaultDiffDirectoryName);
+                    PathUtility.EnsureDirectoryExists(_outputPath);
+
+                    Console.WriteLine("Using --output={0}", _outputPath);
+                }
+
+                bool needTestTree = Benchmarks || DoTestTree;
+
+                if ((_platformPath == null) || (_basePath == null) || (_diffPath == null) || ((_testPath == null) && needTestTree))
+                {
+                    // Try all combinations of build and architecture to find one that has both a Core_Root
+                    // and, if necessary, base and diff builds and tests.
+                    //
+                    // If the user already specified one of --core_root, --base, or --diff, we leave
+                    // that alone, and only try to fill in the remaining ones with an appropriate
+                    // default.
+                    //
+                    // E.g.:
+                    //    test_root: c:\gh\coreclr\bin\tests\Windows_NT.x64.checked
+                    //    Core_Root: c:\gh\coreclr\bin\tests\Windows_NT.x64.checked\Tests\Core_Root
+                    //    base/diff: c:\gh\coreclr\bin\Product\Windows_NT.x64.checked
+
+                    List<string> archList;
+                    if (_arch == null)
                     {
-                        _moniker = _platformName;
+                        archList = new List<string> { "x64", "x86" };
                     }
-                    break;
-                    case "Darwin":
+                    else
                     {
-                        _moniker = "OSX";
+                        archList = new List<string> { _arch };
                     }
-                    break;
-                    default:
+
+                    List<string> buildList;
+                    if (_build == null)
                     {
-                        Console.WriteLine("No platform mapping!");
-                        _moniker = "bogus";
+                        buildList = new List<string> { "checked", "debug" };
```

Comment:
Defaulting build flavors for any of these seems dicey.  I for one almost always have out-of-date trees built of one flavor or another.  I'd go so far as to say we should print out what we're picking and prompt for user input before continuing (maybe with an explicit `--quiet` flag to opt out).